avocado/utils/software_manager/backends/rpm.py

```diff
@@ -242,6 +242,32 @@ def is_valid(package_path):
             return True
         return False
 
+    @staticmethod
+    def extract_from_package(package_path, dest_path=None):
+        """Extracts the package content to a specific destination path.
+
+        :param str package_path: path to the rpm package.
+        :param dest_path: destination path to extract the files. Default it
+                          will be the current directory.
+        :returns: path of the extracted file
+        :returns: the path of the extracted files.
+        :rtype: str
+        """
+        abs_path = os.path.abspath(os.path.expanduser(package_path))
+        dest = dest_path or os.path.curdir
+
+        # If something goes wrong process.run will raise a CmdError exception
+        result = process.run("rpm2cpio {} | cpio -dium -D {}".format(abs_path,
+                                                                     dest),
+                             shell=True)
+        if result.exit_status == 0:
+            return dest
+
+        # But, sometimes the process don't crash, instead exits with a non
+        # zero output
```

Comment:
I'm confused, because `process.run()`, without `ignore_status=True` will always raise an `process.CmdError` exception if `exit_status` is non zero.  I don't see how this code would be executed.

Reply:
Never mind, with `shell=True` this is working fine. So yes, you are right, I can' t reproduce the error that I was getting before.